### PR TITLE
major CI/utils cleanup

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
- [x] last commit to be dropped

[EDIT] most recent re-base and check and all builds passes (except Fedora rawhide [bug filed] and CentOS 8 - new issue, to be fixed): https://github.com/pmem/libpmemobj-cpp/pull/963/checks?check_run_id=1530212528

This change is [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)

### CODE REVIEW

**Review by lukaszstolarczuk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)* status: 0 of 22 files reviewed, 4 unresolved discussions

___
*[doc/RELEASE_STEPS.md, line 9 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLD_ml22OuV-VQ61gz1:-MLD_ml22OuV-VQ61gz2:b-u1mcm5) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/doc/RELEASE_STEPS.md#L9)):*
> ```Markdown
> Make a release locally:
> - for major/minor release:
>   - change version of Docker images (IMG_VER) in gha.yml to $VER
> ```

unfortunately I don't see a good option to propagate IMG_VER by the branch name (especially, in case of pull requests or pushes which happen on forked repos). This way we still need to set this var properly, but I propose to do that while doing the release steps - this way we don't have to merge any conflicts when stable branch is merged back to master.
___
*[utils/docker/pull-or-rebuild-image.sh, line 29 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLDabpD7dkz3j2mlEfn:-MLDabpD7dkz3j2mlEfo:bc07ew0) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/pull-or-rebuild-image.sh#L29)):*
> ```Shell
> source $(dirname ${0})/set-ci-vars.sh
> 
> IMG_VER=${IMG_VER:-latest}
> ```

all scripts handling docker images (build.sh, build-image.sh, push-image.sh and pull-or-rebuild.sh) takes IMG_VER from CI. In case of running them locally the default value (if IMG_VER is not given) is "devel", but in this script I propose to use "latest" tag - this way we can locally just pull some image without previously running image building ("latest" is always available on the container registry, while "devel" is only build locally) 
___
*[utils/docker/pull-or-rebuild-image.sh, line 81 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLD_ZmmD-e02kWSS4UR:-MLD_ZmmD-e02kWSS4US:b-wo96do) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/pull-or-rebuild-image.sh#L81)):*
> ```Shell
> if [[ "${1}" == "rebuild" ]]; then
> 	build_the_image
> 	push_the_image
> ```

with this PR we push the image every time it is rebuilt and we don't require "push the image" step in CI definition, but it **may** lead in some corner cases to faulty image being pushed to the container registry...
___
*[utils/docker/images/build-image.sh, line 35 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLDbWTf2-G5_Vsdj0Qx:-MLDbWTf2-G5_Vsdj0Qy:bzmz0t) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/images/build-image.sh#L35)):*
> ```Shell
> fi
> 
> TAG="${OS}-${OS_VER}-${IMG_VER}"
> ```

in this PR I propose to change "IMG_VER" from being a prefix to being a suffix, it will change the default naming convention on container registry, but I believe it's not a problem, since we have for now only few images on new container registry (GHCR.io). This way is more natural, like other images stored on the Internet

**Review by karczex — CHANGES_REQUESTED** [review]

File: utils/docker/pull-or-rebuild-image.sh [hunk 1]
```diff
@@ -48,14 +49,23 @@ function build_the_image() {
 	popd
 }
 
-# If "rebuild" param is passed to the script, force rebuild
+function pull_the_image() {
```
Comment (karczex) [inline utils/docker/pull-or-rebuild-image.sh:53]: Maybe it's me, but word "the" in function name looks odd.

File: .github/workflows/gha.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ jobs:
       # use org's Private Access Token to log in to GitHub Container Registry
       CONTAINER_REG_USER:   ${{ secrets.GH_CR_USER }}
       CONTAINER_REG_PASS:   ${{ secrets.GH_CR_PAT }}
+      FORCE_IMAGE_ACTION:   ${{ secrets.FORCE_IMAGE_ACTION }}
```
Comment (karczex) [inline .github/workflows/gha.yml:20]: What will happen, when if variable is not defined?

Comment (karczex) [inline .github/workflows/gha.yml:20]: What will happen, if variable is not defined?

File: doc/RELEASE_STEPS.md [hunk 3]
```diff
@@ -14,24 +17,26 @@ Make a release locally:
 - git tag -a -s -m "Libpmemobj-cpp Version $VERSION" $VERSION
 
 Undo temporary release changes:
-- git rm .version
-- git commit -m "common: git versioning"
+- git rm .version && git commit -m "common: git versioning"
```
Comment (karczex) [inline doc/RELEASE_STEPS.md:20]: Maybe we should change convention, and have .version file always present in master branch, and increment it in first commit after release.

**Review by karczex — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)* status: 0 of 22 files reviewed, 4 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[utils/docker/pull-or-rebuild-image.sh, line 29 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLDabpD7dkz3j2mlEfn:-MLNm7cV87-kD_g-r2ju:bhyx52y) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/pull-or-rebuild-image.sh#L29)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

all scripts handling docker images (build.sh, build-image.sh, push-image.sh and pull-or-rebuild.sh) takes IMG_VER from CI. In case of running them locally the default value (if IMG_VER is not given) is "devel", but in this script I propose to use "latest" tag - this way we can locally just pull some image without previously running image building ("latest" is always available on the container registry, while "devel" is only build locally) 
</blockquote></details>

As I already mentioned, we may not removing .version file and increment it in first commit after release, and use here and there in CI.
___
*[utils/docker/images/build-image.sh, line 35 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLDbWTf2-G5_Vsdj0Qx:-MLNm0tU5f77M5pWbc2v:b32v57) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/images/build-image.sh#L35)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

in this PR I propose to change "IMG_VER" from being a prefix to being a suffix, it will change the default naming convention on container registry, but I believe it's not a problem, since we have for now only few images on new container registry (GHCR.io). This way is more natural, like other images stored on the Internet
</blockquote></details>

+1

**Review by karczex — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)* status: 0 of 22 files reviewed, 4 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[utils/docker/pull-or-rebuild-image.sh, line 81 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLD_ZmmD-e02kWSS4UR:-MLNlXVzCC6E_sBXn8Yg:b-3vvn80) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/pull-or-rebuild-image.sh#L81)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

with this PR we push the image every time it is rebuilt and we don't require "push the image" step in CI definition, but it **may** lead in some corner cases to faulty image being pushed to the container registry...
</blockquote></details>

IMO that's OK, probability of scenario of two PRs changing image, merged one after another without image rebuild is low.  
You may just add some comment here.

File: utils/docker/build.sh [hunk 4]
```diff
@@ -80,62 +94,53 @@ if [[ "$command" == "" ]]; then
...
-# do not allocate a pseudo-TTY if we are running on GitHub Actions
-[ ! $GITHUB_ACTIONS ] && TTY='-t' || TTY=''
+# Do not allocate a pseudo-TTY if we are running on GitHub Actions
+[ ! "${GITHUB_ACTIONS}" ] && DOCKER_OPTS="${DOCKER_OPTS} --tty=true"
 
-WORKDIR=/libpmemobj-cpp
-SCRIPTSDIR=$WORKDIR/utils/docker
 
-echo Building on ${OS}-${OS_VER}
+echo Running build on docker: ${OS}-${OS_VER}
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply utils/docker/build.sh]: Done.

File: utils/docker/pull-or-rebuild-image.sh (hunk 1 above)
Comment (lukaszstolarczuk) [reply utils/docker/pull-or-rebuild-image.sh]: Done.

File: .github/workflows/gha.yml (hunk 2 above)
Comment (lukaszstolarczuk) [reply .github/workflows/gha.yml:20]: It's just not set, which should be fine. But, I just remembered this can't be used on PR (secrets are not available), so it will work only on master branch (on pushes)

File: doc/RELEASE_STEPS.md (hunk 3 above)
Comment (lukaszstolarczuk) [reply doc/RELEASE_STEPS.md:20]: If we decide to change it, I believe it should be done in sep. PR

**Review by karczex — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)* status: 0 of 22 files reviewed, 5 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[utils/docker/prepare-for-build.sh, line 27 at r4](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MM0hdUp08uK0Rzy8rh3:-MM0hdUp08uK0Rzy8rh4:b-nfy85m) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/f8fe982ef786cd46b14b2effbe5adc78dbf0e642/utils/docker/prepare-for-build.sh#L27)):*
> ```Shell
> 
> # Change WORKDIR's owner on CI build and prepare tmpfs device
> if [ "${CI_RUN}" == "YES" ]; then
> ```

IMO bash source is like tricky verison of import - especially, that this file is kind of "library". It may provide some functions, setup environment variables or even throw some errors at the moment of sourcing,  but changing  file system state is little confusing, and make "library" less usable. It would be better to enclose it in function and call it when needed.

**Comment by karczex** [comment]
___
*[utils/docker/prepare-for-build.sh, line 27 at r4](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MM0hdUp08uK0Rzy8rh3:-MMGhJGlBr2fH7GR-eqE:b-wy0x5m) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/f8fe982ef786cd46b14b2effbe5adc78dbf0e642/utils/docker/prepare-for-build.sh#L27)):*
<details><summary><i>Previously, karczex wrote…</i></summary><blockquote>

IMO bash source is like tricky verison of import - especially, that this file is kind of "library". It may provide some functions, setup environment variables or even throw some errors at the moment of sourcing,  but changing  file system state is little confusing, and make "library" less usable. It would be better to enclose it in function and call it when needed. 
</blockquote></details>

As discussed offline, it may stay as is.

**Review by lukaszstolarczuk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)* status: 0 of 22 files reviewed, 4 unresolved discussions (waiting on @karczex and @lukaszstolarczuk)

___
*[utils/docker/pull-or-rebuild-image.sh, line 29 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLDabpD7dkz3j2mlEfn:-MMGoEQeEHNsa4Zfjs7_:b-6tkwq0) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/pull-or-rebuild-image.sh#L29)):*
<details><summary><i>Previously, karczex wrote…</i></summary><blockquote>

As I already mentioned, we may not removing .version file and increment it in first commit after release, and use here and there in CI.
</blockquote></details>

comment discussed offline - my change here make sense :wink:
___
*[utils/docker/pull-or-rebuild-image.sh, line 81 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MLD_ZmmD-e02kWSS4UR:-MMGpbaW0nMdjk-tcLoj:b8cdnrx) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/b2a9bdda75889da19e7a2026f746cff8c36a477f/utils/docker/pull-or-rebuild-image.sh#L81)):*
<details><summary><i>Previously, karczex wrote…</i></summary><blockquote>

IMO that's OK, probability of scenario of two PRs changing image, merged one after another without image rebuild is low.  
You may just add some comment here. 
</blockquote></details>

Comment is updated on top of the file

**Review by KFilipek — COMMENTED** [review]
"last commit to be dropped" is not marked in first post, what about it?

> Reviewed 13 of 22 files at r1, 4 of 7 files at r2, 3 of 5 files at r5, 1 of 1 files at r6, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-:-MMuR5V7DLellOEMtf0d:b-ca1tnb)* status: all files reviewed, 10 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[utils/docker/build.sh, line 7 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuSflk5Qhn6HmupXBD:-MMuSfll15-2jK6-12WO:b-qv4s5w) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/build.sh#L7)):*
> ```Shell
> #
> # build-local.sh - runs a Docker container from a Docker image with environment
> #		prepared for running libpmemobj-cpp builds and tests. It uses docker image
> ```

Docker - capital letter
___
*[utils/docker/build.sh, line 13 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuT7qlC5DL1UV63RTn:-MMuT7qlC5DL1UV63RTo:b-kbo2ft) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/build.sh#L13)):*
> ```Shell
> # - set env var 'HOST_WORKDIR' to where the root of this project is on the host machine,
> # - set env var 'OS' and 'OS_VER' properly to a system/docker you want to build this
> #	repo on (for proper values take a look at the list of Dockerfiles at the
> ```

Dockerfiles -> Docker files?
___
*[utils/docker/build.sh, line 30 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuStTF109TLfMuh0UM:-MMuStTF109TLfMuh0UN:b-eob38e) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/build.sh#L30)):*
> ```Shell
> IMAGE_NAME=${CONTAINER_REG}:${TAG}
> CONTAINER_NAME=libpmemobj-cpp-${OS}-${OS_VER}
> WORKDIR=/libpmemobj-cpp  # working dir within docker container
> ```

Docker, line no7 has "Docker container", so I suppose that in other lines should be the same 
___
*[utils/docker/pull-or-rebuild-image.sh, line 12 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuUOI545ZTlSMUTjky:-MMuUOI68LtnsCjMfbV3:bs7pzlx) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/pull-or-rebuild-image.sh#L12)):*
> ```Shell
> #		used locally without building any images).
> #
> # If docker was rebuilt and all requirements are fulfilled (more details in
> ```

Docker
___
*[utils/docker/run-build.sh, line 299 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuWoOuEgweMOb-Wf1_:-MMuWoOuEgweMOb-Wf1a:bedyo1f) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/run-build.sh#L299)):*
> ```Shell
> 
> # Main:
> echo "### Cleaning workspace"
> ```

Delete "### "
___
*[utils/docker/run-coverity.sh, line 56 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuXUMpE2qR77RgiYlD:-MMuXUMpE2qR77RgiYlE:b-cu2897) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/run-coverity.sh#L56)):*
> ```Shell
> echo "Run the Coverity scan"
> 
> # XXX: Patch the Coverity script.
> ```

Is it still true?

**Review by lukaszstolarczuk — COMMENTED** [review]
last commit shows the build of all OSes, if we agree to merge this PR I'll drop it.

> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-:-MN3eNPp-yCPsjdhpKrI:b9eajrh)* status: all files reviewed, 10 unresolved discussions (waiting on @KFilipek and @lukaszstolarczuk)

___
*[utils/docker/build.sh, line 7 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuSflk5Qhn6HmupXBD:-MN3hlfrA31LmGJwyfHb:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/build.sh#L7)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Docker - capital letter
</blockquote></details>

Done.
___
*[utils/docker/build.sh, line 13 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuT7qlC5DL1UV63RTn:-MN3ef7u0JpF8snXBF4A:btljgeg) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/build.sh#L13)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Dockerfiles -> Docker files?
</blockquote></details>

`Dockerfile` is a proper word, ref. https://docs.docker.com/engine/reference/builder/ 
___
*[utils/docker/build.sh, line 30 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuStTF109TLfMuh0UM:-MN3gogw23mMYlQMWXeF:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/build.sh#L30)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Docker, line no7 has "Docker container", so I suppose that in other lines should be the same 
</blockquote></details>

Done.
___
*[utils/docker/pull-or-rebuild-image.sh, line 12 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuUOI545ZTlSMUTjky:-MN3kchV5knW6hfS5TWF:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/pull-or-rebuild-image.sh#L12)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Docker
</blockquote></details>

Done.
___
*[utils/docker/run-build.sh, line 299 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuWoOuEgweMOb-Wf1_:-MN3jMkF01D7Qd95wpM8:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/run-build.sh#L299)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Delete "### "
</blockquote></details>

Done.
___
*[utils/docker/run-coverity.sh, line 56 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuXUMpE2qR77RgiYlD:-MN3f61X0YhxzCNJryi6:b-96hwpa) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/run-coverity.sh#L56)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

Is it still true?
</blockquote></details>

yes, you can see we download the patch right below this comment, and it is (still) a workaround - see the rest of the comment, especially "This change should be reverted when the Coverity script is fixed."

File: doc/RELEASE_STEPS.md (hunk 3 above)
Comment (igchor) [reply doc/RELEASE_STEPS.md:20]: But I believe then we will always have some non-latest version reported by e.g. cmake and pkgconfig file since .version file has priority over git describe, right?

File: .github/workflows/gha.yml [hunk 5]
```diff
@@ -34,8 +35,9 @@ jobs:
         with:
           fetch-depth: 50
 
+      # "pull" or "rebuild" can be passed to a secret FORCE_IMAGE_ACTION to override default action
```
Comment (igchor) [inline .github/workflows/gha.yml:49]: why exactly is this change needed?

Comment (lukaszstolarczuk) [reply .github/workflows/gha.yml:49]: when it happens (and we know from the past it did) that image is missing on the container registry, for some reason, we can easily set this secret using GH interface, and then easily rebuilt the image by re-running failed (master) build, using GHA interface. This way we don't need to make this dummy change with whitespace or something to force rebuild image.

File: doc/RELEASE_STEPS.md (hunk 3 above)
Comment (lukaszstolarczuk) [reply doc/RELEASE_STEPS.md:20]: if we decide to do this change then we could also change cmake file; but I believe we have working solution and it seems to work just fine with rather low cost of adding/removing `.version` file on each release... 

As a comparison, PMDK repo has approach with VERSION file with release tag and GIT_VERSION file with git formatting pattern. Either of this two is always committed... but, again, I believe our approach is quite all right.

**Review by KFilipek — COMMENTED** [review]
> Reviewed 1 of 22 files at r1, 4 of 6 files at r8, 1 of 2 files at r11, 1 of 1 files at r12.
> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963)* status: all files reviewed, 5 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[utils/docker/run-build.sh, line 299 at r7](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-MMuWoOuEgweMOb-Wf1_:-MOByXB_3dIOxXFTaYon:b32x01) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/e6204dca2fbcd0b4a69fcde355644621dcb7374d/utils/docker/run-build.sh#L299)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

xD

**Review by KFilipek — COMMENTED** [review]
Very :lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/963#-:-MOBzyUL50RBWxJJKFOW:bj1izjl)* status: all files reviewed, 5 unresolved discussions (waiting on @lukaszstolarczuk)
